### PR TITLE
Feat/BiGG ids

### PR DESCRIPTION
**Pull request by snmendoz**
### Main improvements in this PR:
I summarize here what I did

1) I changed the compartment identifiers to match those in the BiGG database (http://bigg.ucsd.edu/compartments). In particular, "er" was changed to "r" and "p" to "x"

2) I created four fields, called "metBiGGIDs", "rxnBiGGIDs", "metSIDs" and "rxnRIDs". These are to store BiGG identifiers and the current identifiers of the model (e.g. s_0001, r_0001), respectively. This is needed to transform one version into another in an easier way.

3) I filled cells in metBiGGIDs when a synonym was found in the BiGG database. Otherwise, I kept an empty cell. I did the mapping with MNX and manual curation. I did the mapping in a way that all the metabolites were mapped to non-redundant identifiers, so the length of the array of metabolites was kept constant.

4) I created a script to transform from the current identifiers to BiGG identifiers. This function has two modes of operation. In the first mode, it transforms just metabolites and reactions which have a translation to BiGG identifiers. In the second mode. It transforms the entire model using a .csv file with novel BiGG-like identifiers. This second mode of operation could be useful to create a model with new, BiGG-like identifiers and therefore the generated model can be submitted to the BiGG. Another function to transform the model from BiGG to traditional reactions was also created.

This is a group contribution. Contributions are listed according to the CRedit Taxonomy:
Sebastian Mendoza: Data Curation, Software.
Bas Teusink: Resources, Supervision.

**I hereby confirm that I have:**

- [X] Tested my code with [all requirements](https://github.com/SysBioChalmers/yeast-GEM#required-software---contributor) for running the model
- [X] Selected `devel` as a target branch (top left drop-down menu)
- [X] If needed, asked first in the [Gitter chat room](https://gitter.im/SysBioChalmers/yeast-GEM) about this PR

### CODE REVIEW

File: ModelFiles/yml/yeastGEM.yml [hunk 1]
```diff
@@ -20103,6 +21433,7 @@
     - formula: C21H42NO7P
     - charge: 0
     - annotation: !!omap
+      - bigg.metabolite: 1agpe161[ce]
```
[the remark this answers is not in the document]
Comment (hongzhonglu) [reply ModelFiles/yml/yeastGEM.yml]: I agree with that for the metabolite id mapping, the compartment information should not be included.

**Comment by hongzhonglu** [comment]
@snmendoz when I check the rxnID and BiGG id mapping file, i found some BiGG id (like 
DBGPMAMAMAMAMAMAMAMAMBGBGDA1212131213121616141413, GPMAT4) can not be found from BiGG database. Is that named by yourself? If yes, then we need give a note about where and how the BiGG id is obtained from.

File: ModelFiles/yml/yeastGEM.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (snmendoz) [reply ModelFiles/yml/yeastGEM.yml]: Hi @ChristianLieven and @hongzhonglu .
I will remove compartments. Thanks for the review.

Best,

**Comment by snmendoz** [comment]
Hi, 

I attach six files with the following lists

1) metabolites already in BiGG
2) new metabolites IDS (not in BIGG)
3) same as 1 but without compartments
4) same as 2 but without compartments
5) reactions already in BiGG
6) new reactions IDS (not in BIGG)


[rxns_already_in_bigg.txt](https://github.com/SysBioChalmers/yeast-GEM/files/4645267/rxns_already_in_bigg.txt)
[mets_already_in_bigg.txt](https://github.com/SysBioChalmers/yeast-GEM/files/4645268/mets_already_in_bigg.txt)
[mets_already_in_bigg_without_compartments.txt](https://github.com/SysBioChalmers/yeast-GEM/files/4645269/mets_already_in_bigg_without_compartments.txt)



[new_met_IDS.txt](https://github.com/SysBioChalmers/yeast-GEM/files/4645263/new_met_IDS.txt)
[new_met_IDS_without_compartments.txt](https://github.com/SysBioChalmers/yeast-GEM/files/4645265/new_met_IDS_without_compartments.txt)
[new_rxn_IDS.txt](https://github.com/SysBioChalmers/yeast-GEM/files/4645266/new_rxn_IDS.txt)

**Comment by BenjaSanchez** [comment]
@snmendoz thanks for this! I will resume the work here by separating the lists in the PR based on the lists you just provided, and only adding as annotation to the model the ids that already exist in BiGG. The other ids will be kept for converting the model to BiGG-compliant with a separate script.

File: ModelFiles/yml/yeastGEM.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (BenjaSanchez) [reply ModelFiles/yml/yeastGEM.yml]: compartment info removed in [`cdbd689`](https://github.com/SysBioChalmers/yeast-GEM/pull/188/commits/cdbd6895502004150bedcb05a6e27856f18775ff)

File: ComplementaryData/modelCuration/globalRxnDictionary.csv [hunk 2]
```diff
@@ -0,0 +1,3963 @@
...
+r_3630,PMIP1411802181t_ce_c
+r_3631,PMIP1411812181t_ce_c
+r_3632,PMIB14511602161t_c_n
+r_3633,PMIB14511612161t_c_n
+r_3634,PMIB14511802161t_c_n
+r_3635,PMIB14511812161t_c_n
+r_3636,PMIB14511602181t_c_n
+r_3637,PMIB14511612181t_c_n
+r_3638,PMIB14511802181t_c_n
+r_3639,PMIB14511
```
[the remark this answers is not in the document]
Comment (BenjaSanchez) [reply ComplementaryData/modelCuration/globalRxnDictionary.csv]: reaction compartment notation standardized in [`d35822e`](https://github.com/SysBioChalmers/yeast-GEM/pull/188/commits/d35822e6617365a7139607cdd2d2ae9b2bbb0500) and [`59b6d88`](https://github.com/SysBioChalmers/yeast-GEM/pull/188/commits/59b6d887da0b71d263f40fda057b763d492c231f)

**Comment by BenjaSanchez** [comment]
Thanks for this work @snmendoz. I have re-made the PR to be compliant with the changes suggested by @ChristianLieven and @hongzhonglu. All changes summarized below:

1. The branch had undone changes in the model from a couple of merged PRs (#155 and #178), so I reverted that.
2. One KEGG id was corrected to represent the proper isomer.
3. Deleted 2 duplicated reactions (discussed in #187).
4. Split the BiGG lists in 2 separate ones: ids already existing in BiGG and new ids. Only the existing ones are added as annotation to the model.
5. Removed compartment and copy suffixes from met BiGG ids.
6. Preserved the original compartment convention (`er` and `p`) instead of the BiGG-compliant one (`r` and `x`), for backwards compatibility with other model users. However, the reaction BiGG ids comply with the BiGG standard (including switching `erm` to `rm` for consistency).
7. Redefined new metabolite/reaction ids to be compliant with naming conventions in the field/model, including:
  a. Lipid metabolite and reaction BiGG ids.
  b. SLIME reaction BiGG ids.
  c. Exchange reactions.
  d. Pseudo-metabolites and pseudo-reactions associated to biomass.
  e. Transport reactions.
8. Switched to the development branches of cobratoolbox & RAVEN, as only they can properly store BiGG fields. With that, changes from https://github.com/opencobra/cobratoolbox/pull/1590 (featuring gene SBO terms) come into place. **NOTE** that after this branch is merged and until new versions for those toolboxes are released, any further development in `yeast-GEM` should be conducted with the corresponding toolbox branches (otherwise the BiGG fields will be removed) @feiranl.
9. Added the Matlab script for adding BiGG ids as additional annotation from @snmendoz + used said script to add the annotation to the model. I skipped the other script (switching ids) as it is not feasible to use unless a `.mat` version is saved (which we discourage in this repo).

The only thing missing is functionality for converting the `.xml` to be BiGG compliant, by:

1. Changing the main model ids to be BiGG by reading the annotation + `new_IDs` dictionaries.
2. Changing the compartment notations (`p` -> `x`, `er` -> `r` and `erm` -> `rm`).

This would improve compatibility with memote (PR #162) and caffeine (as mentioned in #212). I think therefore a python implementation would make more sense, by replacing the outdated `loadYeastModel.py` with I/O functions `read_yeast_model` and `write_yeast_model`. @edkerk @hongzhonglu I think it's better to add these features to a separate PR (as this is a bit lengthy already), let me know if you agree with this.

**Pull request merged**